### PR TITLE
Migrate browser folder files to typescript & fix npm build

### PR DESCRIPTION
**Pull request by cgewecke**
#148 

**NB: PR targets #153**

PR
+ ports the`browser/` file set to TS and aligns with new typescript config from #152 
+ "fixes" the webpack build 
+ adds a build step to CI

RlpxPeer depends on two Node platform modules which aren't available in the browser: `net` and `dgram`. These probably need to be resolved to some alias. 

Have configured webpack to ignore that problem for the moment so  `npm run build` runs in some fashion for the monorepo transition.

#108 advises that we add a basic test to check the browser build, perhaps we can address whether or not the bundle is valid as part of that.

### CODE REVIEW

**Comment by holgerd77** [comment]
Side question for my understanding (and since I would find it pretty interesting to apply as well: does this "NB: PR targets..." trigger this "Base automatically changed" message or does this always happen if one targets another branch than `master` and the branch gets merged? 🤔

Haven't worked in such a branch cascade before but am constantly battling with similar situations...

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
       - uses: actions/checkout@v2
       - run: npm install
       - run: npm run lint
+      - run: npm run build
```
Comment (holgerd77) [inline .github/workflows/build.yml:24]: Oh yeah, super useful

**Review by holgerd77 — APPROVED** [review]
Looks good.

**Comment by cgewecke** [comment]
@holgerd77 

> Side question for my understanding (and since I would find it pretty interesting to apply as well: does this "NB: PR targets..." trigger this "Base automatically changed" message or does this always happen if one targets another branch than master and the branch gets merged?

Good question! - it's definitely new right?  

Just made the notes for clarity and think this must be the correct explanation:

> if one targets another branch than master and the branch gets merged

**Pull request merged**